File: library/src/main/java/me/shkschneider/skeleton/java/ExceptionHelper.java
package me.shkschneider.skeleton.java;

public class ExceptionHelper {

    protected ExceptionHelper() {
        // Empty
    }

    public static void uncaughtException(final Callback callback) {
        Thread.setDefaultUncaughtExceptionHandler(new Thread.UncaughtExceptionHandler() {
            @Override
            public void uncaughtException(final Thread thread, final Throwable throwable) {
                if (callback != null) {
                    callback.uncaughtException(throwable);
                }
            }
        });
    }

    public interface Callback {

        void uncaughtException(final Throwable throwable);

    }

}

File: library/src/main/kotlin/me/shkschneider/skeleton/java/ExceptionHelper.kt
package me.shkschneider.skeleton.java

object ExceptionHelper {

    fun uncaughtException(callback: Callback) {
        Thread.setDefaultUncaughtExceptionHandler { _, throwable ->
            callback.uncaughtException(throwable)
        }
    }

    interface Callback {

        fun uncaughtException(throwable: Throwable)

    }

}
